src/gallium/drivers/tegra/tegra_stream.c

```diff
@@ -104,21 +104,16 @@ static int tegra_allocate_cmdbuf(struct drm_tegra *drm,
  * tegra_stream_push().
  */
 
-int tegra_stream_create(struct drm_tegra *drm, struct tegra_stream *stream,
+int tegra_stream_create(struct drm_tegra *drm,
+                        struct drm_tegra_channel *channel,
+                        struct tegra_stream *stream,
                         uint32_t words_num)
 {
-    int ret;
-
-    ret = drm_tegra_channel_open(&stream->channel, drm, DRM_TEGRA_GR2D);
-    if (ret) {
-        fprintf(stderr, "drm_tegra_channel_open() failed %d\n", ret);
-        goto err_open_channel;
-    }
-
     words_num = words_num ? words_num : BUFFER_SIZE_WORDS;
 
     stream->status      = TEGRADRM_STREAM_FREE;
     stream->buffer_size = words_num;
+    stream->channel     = channel;
 
```

Comment:
I think we need to borrow the tegra_stream from opentegra as it has some fixes, 947344446cbf0b40a30e07f33654e5a810c4764a was a mistake.